pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java

```diff
@@ -69,6 +70,7 @@ public class ElasticSearchSink implements Sink<GenericObject> {
     private ObjectMapper sortedObjectMapper;
     private List<String> primaryFields = null;
     private final Base64.Encoder base64Encoder = Base64.getEncoder().withoutPadding();
+    private final Pattern nonPrintableCharactersPattern = Pattern.compile("[\\p{C}]");
```

Comment:
Could you add a comment with the reference for this pattern?

Reply:
btw. with Googling I found https://stackoverflow.com/questions/44034232/undocumented-java-regex-character-class-pc .